### ISSUE TITLE
Suggestion: simplify string representation for module objects

### ISSUE THREAD

**Issue by ikappaki** [basilisp-lang/basilisp#1174]
Hi,

would it be possible to provide a shorter string representation (`repr`) for a module object, primarily when returned by the `import` function?

Currently, the repr includes the module's file path, which can be visually very annonying, especially when displayed in notebooks. For example, importing the pytest module in the REPL returns:
```clojure
asilisp.user=> (import pytest)
<module 'pytest' from 'C:\\src\\basilisp\\.venv\\Lib\\site-packages\\pytest\\__init__.py'>
```

I think it would have been better to have it `repr`d keeping the start of the string only
`<module 'pytest'>`

or simply just the module name
`pytest`

This would improve readability and align better with user expectations in environments like notebooks.

It seems to me this could be achieved by overloading the module's class `__repr__` method, though I'm not entirely sure how to implement this.

For comparison, Clojure imports already use a shorter and cleaner representation, which is easier on the eyes:
```clojure
> clj
Clojure 1.12.0
user=> (import java.util.Date)
java.util.Date
user=> (import (java.util Date Calendar) (java.net URI ServerSocket))
java.net.ServerSocket
```

Thanks

**Comment by ikappaki** [comment]
> This is technically fairly easy to accomplish because Basilisp already has [custom logic for emitting the Lisp representation of existing Python objects we don't control](https://github.com/basilisp-lang/basilisp/blob/cd8fd8a69d7909e564f51d86cf975f9988a303f5/src/basilisp/lang/obj.py#L127-L276).

This is pretty cool indeed 🙌
 
> I'm leaning towards 👎 on this one just because it's changing something that Python is giving us for free and introduces more ambiguity at the REPL (particularly since Python modules and packages are named much more similarly to Clojure namespaces), but I could be convinced.

Could you please provide an example of ambiguity, I can't think of one from the top of my head.

In my opinion, most users would likely never care about seeing or using the path of the imported module. Imports typically occur at the start of a session, where the user expects a clean start. 

In Python, imports are silent by default:
```python
> python
Python 3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>>
```

Adding this long path in the  output clutters the screen, especially in Jupyter notebooks, where showing local module paths looks ugly (IMO again). I don't want to share my notebook displaying the local path of my module import. Example:

![image](https://github.com/user-attachments/assets/d7e06e1a-2fa9-4c9e-a037-7612b8f7b887)

I work around this by adding a `nil` next to the import to suppress the output, which is not ideal, especially when sharing notebooks.

Thanks 🙂

**Comment by chrisrink10** [comment]
> I’m highlighting an aesthetic issue with the representation of the import return value on the REPL. Specifically, the module output includes the module file path, which is largely irrelevant to most users and clutters the output with Python module installation paths.

I just don't really think I agree with this point. I think it is valuable information which I would like to be able to view when debugging. 

> Apologies, but I still don’t fully understand how could this be the case. Could you provide an actual example, even if it’s hypothetical, to help clarify?

The Lisp representation is used in debugging and output and will likely be used by other objects (such as maps or vectors) when inspecting their output.

If you were to place a module of any name in some Basilisp collection (for arguments sake we'll call it `attr`) it will appear as just `attr`: `[attr]`. What is that object without context? A namespace? A module? Something else?

> In terms of import aesthetics across runtimes:
>
>   * require in Clojure/Basilisp returns nil.
>   * import in Clojure returns the fully qualified class name (no paths).
>   * import in Python produces no output.

Given that this is a Python-specific situation, I'm not especially convinced by arguments comparing runtimes. Clojure returns `#object[user$eval134$fn__135 0x2787de58 "user$eval134$fn__135@2787de58"]` for functions, which one could argue is just noise. Should Basilisp match that?

That being said, I could certainly be convinced that the `import` macro should return `nil`, which I think would solve your problem.

**Comment by ikappaki** [comment]
> That being said, I could certainly be convinced that the `import` macro should return `nil`, which I think would solve your problem.

Indeed, this would be the most preferable outcome and I believe It addresses your concerns about shortening the `module` representation. Are there are any further concerns you might still have at this point? thanks

**Comment by chrisrink10** [comment]
No this seems fine to me. Are you able to put up the PR for that?

**Comment by ikappaki** [comment]
> No this seems fine to me. Are you able to put up the PR for that?

Yes, will do. It is not as straightforward as one might think at first--just returning `nil` from the the import function-- because the return value is used internally by the `ns` macro, but nevertheless you can shape the PR during review.

**Comment by chrisrink10** [comment]
> > No this seems fine to me. Are you able to put up the PR for that?
> 
> Yes, will do. It is not as straightforward as one might think at first--just returning `nil` from the the import function-- because the return value is used internally by the `ns` macro, but nevertheless you can shape the PR during review.

I actually don't think it's that complicated. The way it's used in `ns` is just generating the calls, not actually using the results.

**Comment by chrisrink10** [comment]
Fixed in #1177
